src/nodeClient.js

```diff
@@ -1,71 +1,55 @@
 const ethers = require('ethers');
-const { helpers } = require('leap-core');
 const LeapProvider = require('leap-provider');
 
 const erc20abi = require('./erc20abi');
 const { formatHostname, advanceBlocks, sleep } = require('./helpers');
 
 let idCounter = 0;
 
-class Node extends helpers.LeapEthers {
+class Node extends LeapProvider {
   constructor(hostname, jsonrpcPort) {
-    const provider = new LeapProvider(formatHostname(hostname, jsonrpcPort))
-    super(provider);
-
+    super(formatHostname(hostname, jsonrpcPort));
     this.id = idCounter++;
     this.hostname = hostname;
     this.port = jsonrpcPort;
   }
 
   async sendTx(tx) {
-    return this.provider.sendTransaction(tx)
+    return this.sendTransaction(tx)
       .then(tx => 
         Promise.race([
           tx.wait(), 
           new Promise((_, reject) => setTimeout(
             () => reject('Transaction not included in block after 5 secs.'),
-            5000
+            7000
           ))
         ])
       );
   };
 
-  async getBalance(addr) {
-    return this.provider.getBalance(addr).then(res => Number(res));
-  }
-
   async getBlock(val, includeTxs) {
     let method = 'eth_getBlockByNumber';
 
     if (typeof val === 'string' && val.startsWith('0x')) {
       method = 'eth_getBlockByHash';
     }
 
-    return this.provider.send(method, [val, includeTxs]);
-  }
-
-  async getTransaction(...args) {
-    // Do it manually here because we sometimes get 'encoding' errors (missing fields returned from rpc?)
-    return this.provider.send('eth_getTransactionByHash', args);
+    return this.send(method, [val, includeTxs]);
   }
 
   async advanceUntilChange(wallet) {
-    const currentBlock = await this.provider.getBlockNumber();
+    const currentBlock = await this.getBlockNumber();
 
-    let colors = (await this.provider.send('plasma_getColors', [true, false]));
-    colors = colors.concat((await this.provider.send('plasma_getColors', [false, true])));
-    colors = colors.concat((await this.provider.send('plasma_getColors', [false, false])));
+    let colors = Object.values(await this.getColors()).reduce((s, a) => s.concat(a), []);
 
     while (true) {
-      const blockNumber = await this.provider.getBlockNumber();
+      const blockNumber = await this.getBlockNumber();
 
       if (blockNumber > currentBlock) {
         break;
       }
 
-      let c = (await this.provider.send('plasma_getColors', [true, false]));
-      c = c.concat((await this.provider.send('plasma_getColors', [false, true])));
-      c = c.concat((await this.provider.send('plasma_getColors', [false, false])));
+      let c = Object.values(await this.getColors()).reduce((s, a) => s.concat(a), []);
```

Comment:
`Object.values(await this.getColors()).flat()`

Reply:
`TypeError: Object.values(...).flat is not a function` 🤔

Reply:
ah.. not supported in node.js LTS yet